day2/lib/day2.rb

```diff
@@ -0,0 +1,41 @@
+module Day2
+  class PasswordList
+    attr_accessor :password_list, :entry_counts
+
+    def initialize(password_list)
+      @password_list = password_list
+    end
+
+    def self.from_file(filepath)
+      new(File.foreach(filepath).map { |line| line.chomp })
+    end
+
+    def count_valid_entries(entry_klass)
+      @password_list.map { |raw_entry| entry_klass.new(raw_entry).valid? }.count(true)
```

Comment:
Look into `.reduce` here

Reply:
This is what I've ended up with. I can't decide if I like it more or not. Either this:

```ruby
def count_valid_entries(entry_klass)
  @password_list.map do |raw_entry|
    entry_klass.new(raw_entry)
  end.reduce(0) { |count, value| count + (value.valid? ? 1 : 0) }
end
```

Or this:

```ruby
def count_valid_entries(entry_klass)
  @password_list.map do |raw_entry|
    entry_klass.new(raw_entry)
  end.reduce(0) do |count, value|
    count + (value.valid? ? 1 : 0)
  end
end
```

Reply:
Ah, here we go. I think I like this the best:

```ruby
def count_valid_entries(entry_klass)
  @password_list.reduce(0) do |valid_entries, raw_entry|
    valid_entries + (entry_klass.new(raw_entry).valid? ? 1 : 0)
  end
end
```